litmus/director/TCID-DIR-OP-CSP-REC-CREATE-MIRROR/test.yml

```diff
@@ -0,0 +1,193 @@
+            return_content: yes
+            body_format: json
+            status_code: 200
+          register: recommendations
+        
+        ## Fetch the recommendation id 
+        - name: Fetch the recommendation id
+          set_fact:
+            recommendation_id: "{{ recommendations.json.data[0].id }}"
+
+        ## List Capacity Recommendations
+        - name: List  Capacity Recommendations
+          uri:
+            url: '{{ director_url }}/v3/groups/{{ group_id }}/recommendations/{{ recommendation_id }}/?action=getcapacityrecommendation'
+            method: POST
+            url_username: '{{ username.stdout }}'
+            url_password: '{{ password.stdout }}'
+            return_content: yes
+            force_basic_auth: yes
+            body_format: json
+            body: '{"clusterId":"{{ cluster_id }}", "raidGroupConfig":{"groupDeviceCount":2, "type":"mirror"}}'
+            status_code: 201
+          register: recommendation_list
+
+        ## List Device Recommendations
+        - name: List Device Recommendations
+          uri:
+            url: '{{ director_url }}/v3/groups/{{ group_id }}/recommendations/{{ recommendation_id }}/?action=getdevicerecommendation'
+            method: POST
+            url_username: '{{ username.stdout }}'
+            url_password: '{{ password.stdout }}'
+            return_content: yes
+            force_basic_auth: yes
+            body_format: json
+            body: '{"clusterId":"{{ cluster_id }}", "deviceGroupName": null,"poolCapacity":"1G","poolName":null, "raidGroupConfig":{"groupDeviceCount":2, "type":"mirror"}}'
+            status_code: 201
+          register: device_recommendation
+
+        ## Create cstorPoolOperation
+        - name: create cstorpooloperation
+          uri: 
+            url: '{{ director_url }}/v3/groups/{{ group_id }}/cstorpooloperations'
+            method: POST
+            url_username: '{{ username.stdout }}'
+            url_password: '{{ password.stdout }}'
+            return_content: yes
+            force_basic_auth: yes
+            body_format: json
+            body: '{"clusterId":"{{ cluster_id }}", "input":"{{ device_recommendation.json.data[0] }}", "kind":"CreateCStorPoolCluster"}'
+            status_code: 201
+          register: cstorpooloperation
+
+        ## Fetching cstorpooloperation id
+        - name: Fetching cstorpooloperation id
+          set_fact:
+            cstorpooloperation_id: "{{ cstorpooloperation.json.id }}"
+
+        ## Execute CStorPoolOpetation
+        - name: Execute CStorPoolOpetation
+          uri: 
+            url: '{{ director_url }}/v3/groups/{{ group_id }}/cstorpooloperations/{{ cstorpooloperation_id }}/?action=execute'
+            method: POST
+            url_username: '{{ username.stdout }}'
+            url_password: '{{ password.stdout }}'
+            return_content: yes
+            force_basic_auth: yes
+            body_format: json
+            body: '{}'
```

Comment:
who provides the body? is it not required? is the body filled up from some id?

Reply:
Here we have to send body empty no need of any id here @AmitKumarDas